CMakeLists.txt

```diff
@@ -370,56 +350,17 @@ function(add_halide_test TARGET)
   set(oneValueArgs WORKING_DIRECTORY)
   set(multiValueArgs GROUPS)
   cmake_parse_arguments(args "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
-  foreach(GROUP ${args_GROUPS})
-    if(NOT TARGET "${GROUP}")
-      define_test_group("${GROUP}")
-    endif()
-  endforeach()
-
-  set(BUILD_NAME "${TARGET}")
-  set_target_properties("${BUILD_NAME}" PROPERTIES EXCLUDE_FROM_ALL TRUE)
-  add_dependencies(build_tests "${BUILD_NAME}")
 
-  set(EXEC_NAME "run_${TARGET}")
-  get_target_property(TARGET_TYPE "${TARGET}" TYPE)
-  if("${TARGET_TYPE}" STREQUAL "EXECUTABLE")
-    if(${args_EXPECT_FAILURE})
-      set(COMMAND "${HALIDE_BASE_DIR}/test/common/expect_failure.sh" "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${TARGET}")
-    else()
-      set(COMMAND "${TARGET}")
-    endif()
-    add_custom_target("${EXEC_NAME}"
-                     COMMAND ${COMMAND}
-                     WORKING_DIRECTORY ${args_WORKING_DIRECTORY}
-                     COMMENT "Executing ${TARGET}"
-                     DEPENDS "${TARGET}")
-  elseif("${TARGET_TYPE}" STREQUAL "UTILITY")
-    # It's probably a custom target or a group: just depend on it.
-    add_custom_target("${EXEC_NAME}" DEPENDS "${TARGET}")
-  else()
-    message(FATAL_ERROR "add_halide_test(): unsupported type ${TARGET_TYPE} for ${TARGET}")
-  endif()
-  set_target_properties("${EXEC_NAME}" PROPERTIES EXCLUDE_FROM_ALL TRUE)
-  foreach(GROUP ${args_GROUPS})
-    add_dependencies("${GROUP}" "${EXEC_NAME}")
-  endforeach()
-  add_test(NAME ${TARGET} COMMAND ${COMMAND})
+  add_test(NAME ${TARGET}
+          COMMAND ${TARGET}
+          WORKING_DIRECTORY "${args_WORKING_DIRECTORY}")
 
+  set_tests_properties(${TARGET} PROPERTIES LABELS "${args_GROUPS}")
+  if (${args_EXPECT_FAILURE})
+    set_tests_properties(${TARGET} PROPERTIES WILL_FAIL true)
+  endif ()
 endfunction()
 
-# Only groups that are defined in *this file* will have projects reliably
```

Comment:
Does the CTest stuff mean this comment is no longer a concern?

Reply:
I assume so.

Reply:
Reason being that these commands were creating targets at the root of the project, which are visible to CMakeLists in subdirectories. CTest handles all that sort of thing for you.

Reply:
I didn't know CTest existed back when I wrote that explicit test stuff.

Reply:
> I didn't know CTest existed back when I wrote that explicit test stuff.

If you were actually on version 3.3, it might not have!